firefox-ios/Client/Frontend/Browser/StartAtHomeHelper.swift

```diff
@@ -58,8 +58,8 @@ class StartAtHomeHelper: FeatureFlaggable {
             timeSinceLastActivity = dateComponents.hour ?? 0
             timeToOpenNewHome = 4
         } else if setting == .always {
-            timeSinceLastActivity = dateComponents.second ?? 0
-            timeToOpenNewHome = 5
+            // When "always", time to open new home should be immediate, no delay
+            return true
```

Comment:
Given that we're changing this, we should probably clean up the whole function. It could look something like:

```swift
 public func shouldStartAtHome() -> Bool {
        let setting = startAtHomeSetting

        let lastActiveTimestamp = UserDefaults.standard.object(forKey: "LastActiveTimestamp") as? Date ?? Date()
        let dateComponents = Calendar.current.dateComponents([.hour, .second],
                                                             from: lastActiveTimestamp,
                                                             to: Date())

        switch setting {
        case .afterFourHours:
            return 4 >= dateComponents.hour ?? 0
        case .always:
            // When "always", time to open new home should be immediate, no delay
            return true
        case .disabled:
            return false
        }
    }
```

Reply:
its done, i have made the changes